scripts/chatty_news.js

```diff
@@ -0,0 +1,512 @@
+                            color: ${forecolor} !important;
+                            text-decoration: none;
+                        }
+                    `;
+                    if(getSetting("chatty_news_show_image")){
+                        styletext += `
+                            div.article { 
+                                background-color: ${backcolor}; position: relative;
+                            }
+                            div.threads { 
+                                position: relative; 
+                            }
+                            div.threads ul ul li.last, div.commentsblock, body.page-chatty #content { 
+                                background-color: transparent !important;
+                            }
+                            #chatty_comments_wrap { 
+                                padding-bottom: 120%;
+                            }
+                            div div.treeview ul li:last-child { 
+                                background: url(../images/commentsbulletlast.gif) 2px top no-repeat;
+                            }
+                            footer.primary {
+                                position: relative; margin-top: 100px;
+                            }
+                            #main ,content-inner {
+                                background-color: transparent;
+                            }
+                            #chatty_comments_wrap {
+                                background-color: transparent !important;
+                                position: relative;
+                            }
+                            #featured-thread {
+                                position: relative;
+                            }
+                        `;
+                    }
+
+                    if(getSetting("chatty_news_highlight_article_posts")){
+                        styletext+="div.fpauthor_14475 { background-color: #068 !important; border: 1px solid white !important; }\n";
+                    }
+                    
+                    if(document.location.href.indexOf("https")>-1){
+                        chattynews.sendAjax("https://www.shacknews.com/feed/rss",chattynews.parseRSS);
+                    }else{
+                        chattynews.sendAjax("http://www.shacknews.com/feed/rss",chattynews.parseRSS);
+                    }
+                    
+                    var head = document.getElementsByTagName('head')[0];
+                    var style = document.createElement('style');
+                    style.type = 'text/css';
+                    if (style.styleSheet){
+                        style.styleSheet.cssText = styletext;
+                    } else {
+                        style.appendChild(document.createTextNode(styletext));
+                    }
+                    head.appendChild(style);
+                }
+            },
+            sendAjax: function SendAjax(url, callbackDoc, callbackText) {
```

Comment:
We've got some helper functions for what you're doing here. They hang out in `./common.js`: `fetchSafe()` is preferred right now due to auto-sanitization of the XHR to satisfy Mozilla security requirements, but the promisified `xhrRequest()` is a good alternative. The extra boilerplate involved in redefining XMLHttpRequest inside any given script reduces readability and testability and can lead to bugs.